pennylane/circuit_graph.py

```diff
@@ -469,9 +470,57 @@ def get_depth(self):
                 self._operation_graph = self._graph.subgraph(
                     list(self._graph.nodes().index(node) for node in self.operations)
                 )
-                self._depth = rx.dag_longest_path_length(self._operation_graph) + 1
+                self._extend_graph(self._operation_graph)
+                self._depth = (
+                    rx.dag_longest_path_length(
+                        self._operation_graph, weight_fn=lambda _, __, w: self._weight_func(w)
+                    )
+                    + 1
+                )
         return self._depth
 
+    @staticmethod
+    def _weight_func(weight):
+        """If weight is a number, use it!"""
+        if isinstance(weight, (int, float)):
```

Comment:
Just a fly-by comment, you an use the Python `numbers` package (see https://docs.python.org/3/library/numbers.html and https://peps.python.org/pep-3141/) for this to generically check if the type is numeric.

```python
from numbers import Number
...
if isinstance(weight, Number)
```

Reply:
Nice catch!